packages/ts-api-react-forms/src/ApiAction.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
```

Comment:
This whole thing confuses me. According to commit message, the idea is to "use API without forms". So why is this located in `ts-api-react-forms` package? Why not `ts-api-react`, or better yet directly in `thunderstore-api`?

Secondly, what seems to me to basically be a "onChange" function is for some reason formatted as a React component? Why?

This uses zod, which **might** be a good idea, but on the other hand **might** be overkill. It was used on the forms since RHF supports it and helps reduce boilerplate(?). Now were not using RHF. Would regular TS interfaces do the same job?

Also I'm not a fan of the `metaData`, as a variable name nor as a concept. YMMV but if you compare this to Dapper, everything above `thunderstore-api` methods treats data as "everything that is needed to make the request, defined by the ts-api method arguments". It's then up to the method to decide what parts of information goes to URL and what goes to query parameters (or request body, although no such use case has been created by me). This way Cyberstorm components don't (need to) know how the data is used, they're just responsible to providing it all.

In summary: I would cut out out `ts-api-react-forms`/`ts-api-react` and go either directly to `thunderstore-api` or possibly through a separate `ts-api-actions` package. That being said, I feel a lot of the above might be due to me not comprehending the inner workings of `ts-api-react-forms`/`ts-api-react` so you might want a second opinion from @MythicManiac

Reply:
Forgot to elaborate on the naming of `metaData`: It's one word, so `metadata`. Even then I'm not sure if it's really "data about the rest of the data". It's additional data, and in this care very much required to form the complete set of "data".